### ISSUE TITLE
Clicking extension toggles (ActionItems) has long-delayed UI feedback

### ISSUE THREAD

**Issue by felixfbecker** [sourcegraph/sourcegraph#10555]
When you click a toggle like the one contributed by the Codecov extension to toggle line decorations, it doesn't immediately go into the visual "toggled" state. It calls a command in the extension worker, which calls back to make a request to the GraphQL API to update user settings, _upon completion_ the in-memory settings are updated which updates the context, which reevaluates context key expressions, which brings the button into the toggled state.

This is very bad for the feeling of a responsive UI.

My proposal is to optimistically update the in-memory settings (while kicking off the GraphQL request in the background). If the GraphQL request fails, roll back the in-memory settings.

**Comment by Joelkw** [comment]
@felixfbecker is this still relevant given our extensions changes two iterations ago?

**Comment by felixfbecker** [comment]
Yeah and imo a top UX problem to fix with extensions (easy to reproduce - you can click any action item toggle like Codecov or Git blame)
